src/Widgets/TerminalWidget.vala

```diff
@@ -103,6 +102,25 @@ namespace PantheonTerminal {
             private set;
         }
 
+        public double _zoom_factor = 1.0;
+        public double zoom_factor {
+            get {
+                return _zoom_factor;
+            }
+            set {
+                _zoom_factor = value;
+                Pango.FontDescription current_font = this.get_font ();
+                if (current_font != null) {
+                    if (default_size == 0) {
+                        default_size = current_font.get_size ();
+                    }
+
+                    current_font.set_size ((int) Math.floor (default_size * zoom_factor));
+                    this.set_font (current_font);
+                }
+            }
+        }
```

Comment:
Some sanity checking of zoom_factor would be a good idea.

Reply:
That sounds more like another issue, as that problem would have existed before these changes.

Reply:
At the moment you are duplicating some code in the increment, decrement and restore_default zoom functions and these also contain some range checking.  If you simplify these functions to just change the zoom factor then the range checking can be consolidated in the zoom factor setter.

Reply:
I have simplified the increment, decrement and default zoom functions, I think it makes sense to only test the zoom_factor for the range when increment or decrementing as the functions exist and there is no reason to test it otherwise. What do you think a sane range would be? Currently the limits are a maximum of 590% and a minimum of 10%; I think from my testing 40% and 250% seem better limits for what a user is likely to want although I am not sure about the behaviour on HiDPI screens.